tests/cupy_tests/functional_tests/test_vectorize.py

```diff
@@ -0,0 +1,256 @@
+    @testing.numpy_cupy_array_equal(accept_error=TypeError)
+    def test_vectorize_le(self, xp, dtype1, dtype2):
+        def my_le(x, y):
+            return x <= y
+
+        return self._run(my_le, xp, [dtype1, dtype2])
+
+    @testing.for_all_dtypes_combination(names=('dtype1', 'dtype2'))
+    @testing.numpy_cupy_array_equal(accept_error=TypeError)
+    def test_vectorize_gt(self, xp, dtype1, dtype2):
+        def my_gt(x, y):
+            return x > y
+
+        return self._run(my_gt, xp, [dtype1, dtype2])
+
+    @testing.for_all_dtypes_combination(names=('dtype1', 'dtype2'))
+    @testing.numpy_cupy_array_equal(accept_error=TypeError)
+    def test_vectorize_ge(self, xp, dtype1, dtype2):
+        def my_ge(x, y):
+            return x >= y
+
+        return self._run(my_ge, xp, [dtype1, dtype2])
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal(accept_error=TypeError)
+    def test_vectorize_usub(self, xp, dtype):
+        def my_usub(x):
+            return -x
+
+        return self._run(my_usub, xp, [dtype])
+
+
+class TestVectorizeExprs(unittest.TestCase):
+
+    @testing.for_all_dtypes(name='cond_dtype', no_complex=True)
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose()
+    def test_vectorize_ifexp(self, xp, dtype, cond_dtype):
+        def my_ifexp(c, x, y):
+            return x if c else y
+
+        f = xp.vectorize(my_ifexp)
+        cond = testing.shaped_random((20, 30), xp, cond_dtype, seed=0)
+        x = testing.shaped_random((20, 30), xp, dtype, seed=1)
+        y = testing.shaped_random((20, 30), xp, dtype, seed=2)
+        return f(cond, x, y)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose()
+    def test_vectorize_incr(self, xp, dtype):
+        def my_incr(x):
+            return x + 1
+
+        f = xp.vectorize(my_incr)
+        x = testing.shaped_random((20, 30), xp, dtype, seed=0)
+        return f(x)
+
+
+class TestVectorizeInstructions(unittest.TestCase):
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose()
+    def test_vectorize_assign_new(self, xp, dtype):
+        def my_assign(x):
+            y = x + x
+            return x + y
+
+        f = xp.vectorize(my_assign)
+        x = testing.shaped_random((20, 30), xp, dtype, seed=1)
+        return f(x)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose()
+    def test_vectorize_assign_update(self, xp, dtype):
+        def my_assign(x):
+            x = x + x
+            return x + x
+
+        f = xp.vectorize(my_assign)
+        x = testing.shaped_random((20, 30), xp, dtype, seed=1)
+        return f(x)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_allclose()
+    def test_vectorize_augassign(self, xp, dtype):
+        def my_augassign(x):
+            x += x
+            return x + x
+
+        f = xp.vectorize(my_augassign)
+        x = testing.shaped_random((20, 30), xp, dtype, seed=1)
+        return f(x)
```

Comment:
How about testing functions with multiple operations and constants?
`return x+x*x+2` or something similar so we can get an ast with a bit of depth :)